src/managed/Microsoft.Extensions.DependencyModel/UnifiedJsonReader.Utf8JsonReader.cs

```diff
@@ -160,8 +146,13 @@ public bool ReadAsBoolean(bool defaultValue)
 
         private static Exception CreateUnexpectedException(ref Utf8JsonReader reader, string expected)
         {
+            // Replace with public API once https://github.com/dotnet/corefx/issues/34768 is fixed
+            object boxedState = reader.CurrentState;
+            long lineNumber = (long)(typeof(JsonReaderState).GetField("_lineNumber")?.GetValue(boxedState) ?? -1);
+            long bytePositionInLine = (long)(typeof(JsonReaderState).GetField("_bytePositionInLine")?.GetValue(boxedState) ?? -1);
```

Comment:
I'm assuming this will go away naturally when the abstraction is removed for https://github.com/dotnet/core-setup/issues/5045, otherwise maybe we should track it separately.

Reply:
It won't go away with #5045, it'll need an issue to track removal.  I'll file one.